src/Rules/Operators/OperatorRuleHelper.php

```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Operators;
+
+use PHPStan\Type\FloatType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\Type;
+
+class OperatorRuleHelper
+{
+
+	public static function isValidForArithmeticOperation(Type $leftType, Type $rightType): bool
+	{
+		if (!$leftType instanceof IntegerType && !$leftType instanceof FloatType) {
```

Comment:
This needs to use `isSubtypeOf` instead of `instanceof`.

Reply:
Sorry, I meant `isSuperTypeOf` (`isSubTypeOf` is mostly internal helper used for *inversion of control*). `accepts()` will also work if you want to allow `mixed`.

> I tried (…) got bunch of internal errors

That's odd. The code looks fine. What errors did you get?

Reply:
It was only happening during tests, didn't find you how to enable debug mode for them...

Anyway updated to isSupersetOf, could you review now?